### PR TITLE
feat: add authentication

### PR DESCRIPTION
**Pull request by anik31**
## Authentication
Added:
- sign in
- sign up
- logout
 
## Files to check
- src/components/Navbar/Navbar.jsx
- src/components/RequireAuth.js
- src/context/auth-context.js
- src/pages/auth/Login.jsx
- src/pages/auth/Signup.jsx
 
 _PS - Ignore other files._

https://user-images.githubusercontent.com/56336326/162269043-6b189f47-b42f-4721-8734-566415ed8d90.mp4

### CODE REVIEW

**Review by sainath7878 — APPROVED** [review]
Nothing to suggest. Code is neat and clean!

File: src/pages/auth/Signup.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            <input type="text" value={credentials.firstName} placeholder="First name" onChange={(e)=>setCredentials(prev=>({...prev, firstName: e.target.value}))} />
+        </div>
+        <div className="input input-text">
+            <label>Last name</label>
+            <input type="text" value={credentials.lastName} placeholder="Last name" onChange={(e)=>setCredentials(prev=>({...prev, lastName: e.target.value}))} />
+        </div>
+        <div className="input input-text">
+            <label>Password</label>
+            <div className="password-wrapper">
+            <input type={isPasswordVisible?"text":"password"} value={credentials.password} placeholder="password" onChange={(e)=>setCredentials(prev=>({...prev, password: e.target.value}))} />
```
Comment (Maianki) [inline src/pages/auth/Signup.jsx:53]: Nitpick: You can extract callback for onChange

**Review by Maianki — APPROVED** [review]
Everything looks good to me. You can go ahead with the merge. Good job

File: src/pages/auth/Signup.jsx (hunk 1 above)
Comment (anik31) [reply src/pages/auth/Signup.jsx:53]: okay will do it after submission

**Pull request merged**